File: sdk-py/getpatter/llm/cerebras.py
"""Cerebras LLM for Patter pipeline mode."""

from __future__ import annotations

import os

from getpatter.providers.cerebras_llm import CerebrasLLMProvider as _CerebrasLLM

__all__ = ["LLM"]


class LLM(_CerebrasLLM):
    """Cerebras LLM provider (OpenAI-compatible Inference API).

    Example::

        from getpatter.llm import cerebras

        llm = cerebras.LLM()                          # reads CEREBRAS_API_KEY
        llm = cerebras.LLM(api_key="csk-...", model="llama3.1-8b")
    """

    def __init__(
        self,
        api_key: str | None = None,
        *,
        model: str = "llama3.1-8b",
        **kwargs,
    ) -> None:
        key = api_key or os.environ.get("CEREBRAS_API_KEY")
        if not key:
            raise ValueError(
                "Cerebras LLM requires an api_key. Pass api_key='csk-...' or "
                "set CEREBRAS_API_KEY in the environment."
            )
        super().__init__(api_key=key, model=model, **kwargs)

File: sdk-ts/src/llm/cerebras.ts
/** Cerebras LLM for Patter pipeline mode. */
import { CerebrasLLMProvider as _CerebrasLLM } from "../providers/cerebras-llm";

export interface CerebrasLLMOptions {
  /** API key. Falls back to CEREBRAS_API_KEY env var when omitted. */
  apiKey?: string;
  /** Model id (e.g. ``"llama-3.3-70b"``). */
  model?: string;
  /** Override the OpenAI-compatible base URL (rarely needed). */
  baseUrl?: string;
  /** Gzip request payloads for faster TTFT on large prompts. */
  gzipCompression?: boolean;
  /** Sampling temperature [0, 2]. */
  temperature?: number;
  /** Max tokens in the assistant response. */
  maxTokens?: number;
}

/**
 * Cerebras LLM provider (OpenAI-compatible Inference API, streaming).
 *
 * @example
 * ```ts
 * import * as cerebras from "getpatter/llm/cerebras";
 * const llm = new cerebras.LLM();                              // reads CEREBRAS_API_KEY
 * const llm = new cerebras.LLM({ apiKey: "csk-...", model: "llama-3.3-70b" });
 * ```
 */
export class LLM extends _CerebrasLLM {
  static readonly providerKey = "cerebras";
  constructor(opts: CerebrasLLMOptions = {}) {
    const key = opts.apiKey ?? process.env.CEREBRAS_API_KEY;
    if (!key) {
      throw new Error(
        "Cerebras LLM requires an apiKey. Pass { apiKey: 'csk-...' } or set CEREBRAS_API_KEY.",
      );
    }
    super({
      apiKey: key,
      model: opts.model,
      baseUrl: opts.baseUrl,
      gzipCompression: opts.gzipCompression,
      temperature: opts.temperature,
      maxTokens: opts.maxTokens,
    });
  }
}
